draft-irtf-cfrg-vdaf.md

```diff
@@ -3153,107 +3162,106 @@ struct {
-linear operations on (its share of) the measurement and proof.
+Our proof system is much the same, except the verifier is split across multiple
+Aggregators, each of which has a secret share of the measurement rather than an
+encryption of it.
 
-To make this work, the proof system is restricted to validity circuits that
-exhibit a special structure. Specifically, an arithmetic circuit with "G-gates"
-(see {{BBCGGI19}}, Definition 5.2) is composed of affine gates and any number of
-instances of a distinguished gate `G`, which may be non-affine. We will refer to
-this class of circuits as 'gadget circuits' and to `G` as the "gadget".
+Validity is defined in terms of an arithmetic circuit evaluated over the
+measurement. The inputs to this circuit are elements of a finite field that
+comprise the encoded measurement; the gates of the circuit are multiplication,
+addition, and negation operations; and the output of the circuit is a single
+field element. If the value is zero, then the measurement is deemed valid;
+otherwise, if the output is non-zero, then the measurement is deemed invalid.
 
-As an illustrative example, consider a validity circuit `C` that recognizes the
-set `L = set([0], [1])`. That is, `C` takes as input a length-1 vector `x` and
-returns 0 if `x[0]` is in `[0,2)` and outputs something else otherwise. This
-circuit can be expressed as the following degree-2 polynomial:
+For example, the simplest circuit specified in this document is the following
+({{prio3count}}):
 
 ~~~
-C(x) = (x[0] - 1) * x[0] = x[0]^2 - x[0]
+C(x) = x * (x-1)
 ~~~
 
-This polynomial recognizes `L` because `x[0]^2 = x[0]` is only true if `x[0] ==
-0` or `x[0] == 1`. Notice that the polynomial involves a non-affine operation,
-`x[0]^2`. In order to apply {{BBCGGI19}}, Theorem 4.3, the circuit needs to be
-rewritten in terms of a gadget that subsumes this non-affine operation. For
-example, the gadget might be multiplication:
+This circuit contains one negation gate (`-1`), one addition gate (`x + (-1)`),
+and one multiplication gate (`x * (x + (-1))`). Observe that `C(x) = 0` if and
+only if `x in range(2)`.
+
+Our goal is to allow each Aggregator, who holds a secret share of `x`, to
+correctly compute a secret share of `C(x)`. This allows the Aggregators to
+determine validity by combining their shares of the output.
+
+Suppose for a moment that the validity circuit `C` is affine, meaning its only
+operations are negation, addition, and multiplication-by-constant. (The circuit
+above is non-affine because it contains a multiplication gate with non-constant
+inputs.) Then each Aggregator can compute its share locally, since
 
 ~~~
-Mul(left, right) = left * right
+C(x_shares[0] + ... + x_shares[SHARES-1]) =
+    C(x_shares[0]) + ... + C(x_shares[SHARES-1])
 ~~~
 
-The validity circuit can then be rewritten in terms of `Mul` like so:
+(Note that, for this equality to hold, it is necessary to scale any constants
```

Comment:
Constants used in constant multiplications are fine as-is, but constants that get added need to be scaled, since this is one way of replacing them with additive shares of themselves.
```suggestion
(Note that, for this equality to hold, it is necessary to scale any addition of a constant
```

Reply:
Good catch.